### PR TITLE
fix(route): Fix xiaoheihe

### LINKED ISSUE #11511: 小黑盒游戏新闻失效
### 路由地址

```routes
/xiaoheihe/news
```

### 完整路由地址

```fullroutes
/xiaoheihe/news
```

### 相关文档

https://docs.rsshub.app/game.html#xiao-hei-he

### 预期是什么？

正常获取内容

### 实际发生了什么？

Error message: response.data.result.filter is not a function

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

_No response_

### 额外信息

```shell
Route requested: /xiaoheihe/news
Error message: response.data.result.filter is not a function
Helpful Information to provide when opening issue:
Path: /xiaoheihe/news
Node version: v16.19.0
Git Hash: 0cdbb26
```

### 这不是重复的 issue

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by tssujt**
## Involved Issue / 该 PR 相关 Issue

Close #11511

## Example for the Proposed Route(s) / 路由地址示例

```routes
/xiaoheihe/news
/xiaoheihe/discount/pc
/xiaoheihe/discount/psn
/xiaoheihe/discount/switch
/xiaoheihe/discount/xbox
/xiaoheihe/user/30664023
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

Fix deprecated routes for xiaoheihe

### CODE REVIEW

File: lib/routes/xiaoheihe/discount.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { Route } from '@/types';
+import got from '@/utils/got';
+
+export const route: Route = {
+    path: '/discount/:platform',
+    categories: ['game'],
+    example: 'xiaoheihe/discount/pc',
```
Comment (TonyRL) [inline lib/routes/xiaoheihe/discount.ts:7]:
```suggestion
    example: '/xiaoheihe/discount/pc',
```

File: lib/routes/xiaoheihe/discount.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    parameters: { platform: '平台分类，见下表' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
```
Comment (TonyRL) [inline lib/routes/xiaoheihe/discount.ts:16]:
```suggestion
```

File: lib/routes/xiaoheihe/news.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                await page.goto(item.link, {
+                    waitUntil: 'domcontentloaded',
+                });
+                await page.waitForSelector('.article-content');
+
+                const html = await page.evaluate(() => document.documentElement.innerHTML);
+                await page.close();
+                const $ = load(html);
+                item.description = $('.article-content').html();
+                return item;
```
Comment (TonyRL) [inline lib/routes/xiaoheihe/news.ts:52]: Puppeteer is not really required.
You can use https://github.com/huandu/heybox-url to calculate appropriate `hkey` and `nonce` to obtain the API response at `/bbs/app/api/share/data`

File: lib/routes/xiaoheihe/user.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    categories: ['game'],
+    example: '/xiaoheihe/user/30664023',
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
```
Comment (TonyRL) [inline lib/routes/xiaoheihe/user.ts:19]:
```suggestion
```

File: lib/routes/xiaoheihe/news.ts (hunk 3 above)
Comment (tssujt) [reply lib/routes/xiaoheihe/news.ts:52]: I think using Puppeteer will make the availability higher.

> Puppeteer is not really required. You can use [huandu/heybox-url](https://github.com/huandu/heybox-url) to calculate appropriate `hkey` and `nonce` to obtain the API response at `/bbs/app/api/share/data`

Comment (TonyRL) [reply lib/routes/xiaoheihe/news.ts:52]: There are way too many people who don't know they need to deploy Puppeteer or how to deploy it properly. #14812

Comment (tssujt) [reply lib/routes/xiaoheihe/news.ts:52]: @TonyRL When I try to add [huandu/heybox-url](https://github.com/huandu/heybox-url) package, I get the error `Cannot find module 'heybox-url'`, it seems to be a issue with [huandu/heybox-url](https://github.com/huandu/heybox-url). Is it acceptable if I add some of the source code used in [huandu/heybox-url](https://github.com/huandu/heybox-url) into xiaoheihe/util.ts ?

Comment (TonyRL) [reply lib/routes/xiaoheihe/news.ts:52]: Sure, since it's [MIT](https://github.com/huandu/heybox-url/blob/main/LICENSE) licensed.

Comment (tssujt) [reply lib/routes/xiaoheihe/news.ts]: @TonyRL Updated without Puppeteer. PTAL.

File: lib/routes/xiaoheihe/user.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import { Route } from '@/types';
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+import { calculate } from './util';
+
+export const route: Route = {
+    path: '/user/:id',
```
Comment (TonyRL) [inline lib/routes/xiaoheihe/user.ts:8]: Missing parameter description `parameters: { id: 'Something' },`

Comment (tssujt) [reply lib/routes/xiaoheihe/user.ts:8]: @TonyRL Updated

**Pull request merged**